cargo-near-build/src/types/cargo/metadata.rs

```diff
@@ -70,12 +77,17 @@ impl CrateMetadata {
 fn get_cargo_metadata(
     manifest_path: &ManifestPath,
     no_locked: bool,
+    cargo_target_dir: Option<&buildtime_env::CargoTargetDir>,
 ) -> eyre::Result<(cargo_metadata::Metadata, Package)> {
     tracing::info!("Fetching cargo metadata for {}", manifest_path.path);
     let mut cmd = MetadataCommand::new();
     if !no_locked {
         cmd.other_options(["--locked".to_string()]);
     }
+    if let Some(target_dir) = cargo_target_dir {
+        let (key, value) = target_dir.entry();
+        cmd.env(key, value);
```

Comment:
this helps resolve output destination to `env!(CARGO_TARGET_DIR)/near/<package>/<package>.wasm`
similarly to what would've been done in case of `std::env::set_var` before building